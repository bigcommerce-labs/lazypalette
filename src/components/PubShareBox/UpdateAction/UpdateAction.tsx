import React, {SFC} from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';
import { ButtonWrapper } from '../styles';

interface PreviewActionProps {
    handlePublish(): void;
}

const UpdateAction: SFC<PreviewActionProps> = props => {
    return (
        <ButtonWrapper>
            <ButtonInput
                onClick={props.handlePublish}
                classType="primary"
                testId="apply-update"
            >
                Apply Update
            </ButtonInput>
        </ButtonWrapper>
    );
};

export default UpdateAction;
