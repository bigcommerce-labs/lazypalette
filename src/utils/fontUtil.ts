export function parseFont(font: string) {
    const provider = font.split('_')[0];
    switch (provider) {
        case 'Google':
            return parseGoogleFont(font);
        default:
            return null;
    }
}

function parseGoogleFont(font: string) {
    const split = font.split('_');
    const family = split[1];
    let formattedFont = '';
    let weight = split[2];

    if (split.length === 2) {
        formattedFont += `${family}|`;
    } else if (split.length > 2) {
        weight = weight.split(',')[0];
        formattedFont += `${family}:${weight}|`;
    }

    return `//fonts.googleapis.com/css?family=${formattedFont}`;
}
