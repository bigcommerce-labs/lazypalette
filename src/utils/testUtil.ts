export const sel = (id: string) => `[data-test-id="${id}"]`;
