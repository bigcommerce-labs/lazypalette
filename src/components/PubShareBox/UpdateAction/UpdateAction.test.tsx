import { mount, shallow } from 'enzyme';
import React from 'react';

import { sel } from '../../../utils/testUtil';

import UpdateAction from './UpdateAction';

describe('UpdateAction', () => {

    const applyUpdateButton = sel('apply-update');

    const mockHandlePublish = jest.fn();

    const UpdateActionElement = <UpdateAction
        handlePublish={mockHandlePublish}
    />;

    it('renders', () => {
        const inactiveAction = shallow(UpdateActionElement);

        expect(inactiveAction).toMatchSnapshot();
    });

    it('should render "Apply Update" button', () => {
        const inactiveAction = mount(UpdateActionElement);

        expect(inactiveAction.find(applyUpdateButton).hostNodes().length).toBe(1);
    });
});
