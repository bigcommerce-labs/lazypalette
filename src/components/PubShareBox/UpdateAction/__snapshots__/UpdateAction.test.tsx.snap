// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateAction renders 1`] = `
<styled.div>
  <Button
    actionType="primary"
    buttonStyle="contained"
    onClick={[MockFunction]}
    testId="apply-update"
    type="button"
  >
    Apply Update
  </Button>
</styled.div>
`;
